src/auth/auth-manager.ts

```diff
@@ -1,62 +1,86 @@
+import {
+  AccountInfo,
+  InteractionRequiredAuthError,
+  PublicClientApplication,
+} from '@azure/msal-browser';
 import { jwtDecode } from 'jwt-decode';
+import { msalConfig } from '../api/constants';
 import { TokenType } from '../enum';
-import {
-  AccessTokenDecodeType,
-  AuthenticationResponseType,
-  RequestUser,
-} from '../types';
-import { login, refreshSession } from '../api/api';
-
-export const logoutUser: Function = () => {
-  Object.keys(TokenType).forEach((token) =>
-    window.localStorage.removeItem(token)
-  );
-};
+import { tokenDecodeType, AuthenticationResponseType } from '../types';
 
-export const saveUserDataLocalStorage: Function = ({
-  id_token,
-  access_token,
-  refresh_token,
-}: AuthenticationResponseType) => {
-  if (refresh_token) {
-    window.localStorage.setItem(TokenType.refresh, refresh_token);
-  }
-  window.localStorage.setItem(TokenType.bearer, id_token);
-  window.localStorage.setItem(TokenType.access, access_token);
-};
+export const AuthManager = () => {
+  let msalInst: PublicClientApplication | null = null;
 
-export const getToken: Function = (typeToken: TokenType) => {
-  return window.localStorage.getItem(typeToken);
-};
+  const saveUserDataLocalStorage: Function = (token: AuthenticationResponseType) => {
+    window.localStorage.setItem(TokenType.idToken, token?.idToken);
+    window.localStorage.setItem(TokenType.accessToken, token?.accessToken);
+  };
 
-export const isValidAccessToken: Function = () => {
-  const token = getToken(TokenType.access);
-  if (!token) {
-    return false;
-  }
-  const accessTokenDecode: AccessTokenDecodeType = jwtDecode(token);
-  const expDate = new Date(accessTokenDecode?.exp * 1000);
-  return expDate > new Date();
-};
+  const getAccessToken: Function = () => {
+    return window.localStorage.getItem(TokenType.accessToken);
+  };
 
-export const refreshUserSession: Function = async () => {
-  try {
-    const refreshToken: string | null = getToken(TokenType.refresh);
-    if (refreshToken) {
-      const res: AuthenticationResponseType = await refreshSession(
-        refreshToken
-      );
-      saveUserDataLocalStorage({ ...res });
-      return true;
+  const getIdToken: Function = () => {
+    return window.localStorage.getItem(TokenType.idToken);
+  };
+
+  const loginUser = async () => {
+    const result = await msalInst?.loginPopup();
+    saveUserDataLocalStorage(result);
+  };
+
+  const logoutUser: Function = async () => {
+    const currentAccount: AccountInfo | undefined =
+      msalInst?.getAllAccounts()[0];
```

Comment:
Getting active account could be extracted in separate function. Then you could call it here and in refresh function.